SCRUM-21 / SCRUM-24: Initial gateway setup with dotenv config and SES email service integration

prdai opened the pull request.

prdai committed:
Add dotenv configuration and update environment variables; enhance .gitignore

YasogaN commented on index.ts:
```diff
@@ -1,12 +1,15 @@
 
-app.get('/', (req, res) => {
-  res.send('Hello World!');
+app.get(`/api/v${apiVersion}/notification/`, (req, res) => {
```
**Changes Requested:**
Remove trailing slashes from route paths as they are treated as distinct routes. Consider implementing middleware to normalize incoming URLs (e.g., redirect or rewrite) to ensure consistent routing behavior.

**Additional Comments:**
For single-version APIs, static routing like `/api/v1/notification` is perfectly fine. However, if you plan to support multiple API versions or maintain backwards compatibility, it’s better to use a dynamic route parameter such as `/api/:version/notification` for more flexible version handling.

prdai replied:
due to this being not related to the current ticket we are working on, since its related to the gateway service I created a new ticket [SCRUM-42](https://iit-team-kfyvfwj3.atlassian.net/browse/SCRUM-42) to resolve this issue, so we will take this up soon.

Commit:
feat: Implement SES email service with logging and email constants

- Added AWS SES client configuration in src/config/aws/ses.ts
- Introduced a logger using Winston in src/config/logger.ts
- Created email constants for default source email in src/constants/emailConstants.ts
- Added a placeholder for email queue implementation in src/queues/emailQueue.ts
- Defined IEmailService interface in src/services/emailService/IEmailService.ts
- Developed SESEmailService class for sending emails via AWS SES in src/services/emailService/SESEmailService.ts
- Exported email service interfaces and implementations in src/services/emailService/index.ts
- Updated main services index to include email services in src/services/index.ts
- Defined EmailBody type in src/types/email.ts

prdai committed:
fix: Update default source email address and remove SES email service endpoint

YasogaN commented on src/services/emailService/IEmailService.ts:
Shouldn't this go to types?

YasogaN commented on index.ts:
```diff
@@ -1,10 +1,13 @@
 
-app.get('/', (req, res) => {
-  res.send('Hello World!');
+app.get(`/api/v${apiVersion}/notification/`, (req, res) => {
```
Remove trailing slash and consider implementing middleware to normalize URLs  per original comment.
